### ISSUE TITLE
time out when connecting to mssql

### ISSUE THREAD

**Issue by wfreeman** [mage-ai/mage-ai#4664]
hi, all:
my mage version:[v0.9.63](https://www.mage.ai/changelog)
and it's deployed on docker

and i have a  sqlserver:
version: 
------------------------------------------------------------------------------
Microsoft SQL Server 2014 (SP3) (KB4022619) - 12.0.6024.0 (X64) 
Sep  7 2018 01:37:51 
Copyright (c) Microsoft Corporation
Enterprise Edition: Core-based Licensing (64-bit) on Windows NT 6.3 <X64> (Build 9600: )
------------------------------------------------------------------------------

everytime when i try to connect mssql via sql data-loader, timeout error happens:
------------------------------------------------------------------------------
MSSQL initialized
└─ Opening connection to MSSQL database...
---------------------------------------------------------------------------
OperationalError                          Traceback (most recent call last)

File /usr/local/lib/python3.10/site-packages/mage_ai/io/mssql.py:62, in MSSQL.open(self)
     60 def open(self) -> None:
     61     with self.printer.print_msg('Opening connection to MSSQL database'):
---> 62         self._ctx = pyodbc.connect(
     63             self.connection_string,
     64         )
OperationalError: ('HYT00', '[HYT00] [Microsoft][ODBC Driver 18 for SQL Server]Login timeout expired (0) (SQLDriverConnect)')

---------------------------------------------------------------------------
but when i use python dataloader, like polars .etc, it works fine.

what's the problem here, and how can i trace it ?

**Comment by wangxiaoyou1993** [comment]
@wfreeman what python code works for you?

**Comment by wfreeman** [comment]
use polars, based on connectorx i think.
====================================================
    url = "mssql://xxx:bbbb@172.16.55.98:1081/?&charset=utf8&tds_version=7.0"
    import polars as pl
    return pl.read_database(query=query, connection=url).to_pandas()
====================================================

> @wfreeman what python code works for you?

**Comment by wangxiaoyou1993** [comment]
> # use polars, based on connectorx i think.
> ```
> url = "mssql://xxx:bbbb@172.16.55.98:1081/?&charset=utf8&tds_version=7.0"
> import polars as pl
> return pl.read_database(query=query, connection=url).to_pandas()
> ```
> 
> ====================================================
> 
> > @wfreeman what python code works for you?

Got it. You're not using ODBC driver.

**Comment by wfreeman** [comment]
> > # use polars, based on connectorx i think.
> > ```
> > url = "mssql://xxx:bbbb@172.16.55.98:1081/?&charset=utf8&tds_version=7.0"
> > import polars as pl
> > return pl.read_database(query=query, connection=url).to_pandas()
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ====================================================
> > > @wfreeman what python code works for you?
> 
> Got it. You're not using ODBC driver.

yes,  i mean when i use other than ODBC ,  it works fine, what's the problem would be?

**Comment by Zoltan21** [comment]
Is there an answer for this, cause I have the same issue :(
I checked the mage ai odbc drivers, but it has just: ODBC Driver 18 for SQL Server. In my simple python code I used the SQL Server odbc driver which works fine. 
How can I add this 'SQL Server' odbc driver to the mage ai?

**Comment by wangxiaoyou1993** [comment]
> Is there an answer for this, cause I have the same issue :( I checked the mage ai odbc drivers, but it has just: ODBC Driver 18 for SQL Server. In my simple python code I used the SQL Server odbc driver which works fine. How can I add this 'SQL Server' odbc driver to the mage ai?

You can install your driver in the docker image and specify it in your config.

**Comment by Zoltan21** [comment]
@wangxiaoyou1993 Can you please give some hint/steps on how to do this? Sorry I am new to docker

**Comment by wangxiaoyou1993** [comment]
> @wangxiaoyou1993 Can you please give some hint/steps on how to do this? Sorry I am new to docker

You can check this doc: https://docs.mage.ai/integrations/databases/MicrosoftSQLServer#dependencies
use `mageai/mageai:latest` as the base image and build your custom docker image.

**Comment by Zoltan21** [comment]
So I did this, and I created a new docker image with the following:
`# Use the existing mage-ai image as the base image
FROM mageai/mageai

# Install dependencies
RUN apt-get update && apt-get install -y apt-transport-https curl gnupg

# Add the Microsoft SQL Server ODBC Driver repository
RUN curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - \
    && curl https://packages.microsoft.com/config/debian/10/prod.list > /etc/apt/sources.list.d/mssql-release.list

# Install the ODBC Driver for SQL Server
RUN apt-get update \
    && ACCEPT_EULA=Y apt-get install -y msodbcsql17 \
    # && ACCEPT_EULA=Y apt-get install -y msodbcsql18 \
    # && ACCEPT_EULA=Y apt-get install -y mssql-tools18 \
    # && ACCEPT_EULA=Y apt-get install msodbcsql \
    && apt-get install -y unixodbc-dev \
    && rm -rf /var/lib/apt/lists/*

# Set the command to run the application
CMD ["/bin/sh", "-c", "/app/run_app.sh"]`

and I verified that both drivers are available. Unfortunately this still not working:
OperationalError: ('HYT00', '[HYT00] [Microsoft][ODBC Driver 17 for SQL Server]Login timeout expired (0) (SQLDriverConnect)')

**Comment by Zoltan21** [comment]
Then I thought maybe the ODBC Driver 13 for SQL Server could work, but since in the mage ai image the debian 10 is used I cannot use the 13, that is only supported with Debian 8
